src/ostree/repo.h

```diff
@@ -0,0 +1,39 @@
+#ifndef REPO_H
+#define REPO_H
+
+#include <string>
+#include <memory>
+
+#include <ostree.h>
+
+
+namespace OSTree {
+
+class Repo {
+ public:
+  using Ptr = std::shared_ptr<Repo>;
+
+ public:
+  static Ptr create(std::string path) { return std::make_shared<Repo>(std::move(path)); }
```

Comment:
i'm not too up on c++ idioms. This seemed odd to me though. Seems like a user can use the make_shared/make_unique calls directly? The reason i noticed, was that I'd think this might be a std::unique instead of std::shared pointer.

Reply:
Whether unique and shared pointer is needed really depends on an use-case, but in this particular case, we don't need this factory method at all, going to remove it.